EarlGrey/Core/GREYAutomationSetup.m

```diff
@@ -45,37 +45,25 @@ - (void)perform {
   [selfClass grey_setupCrashHandlers];
   // Force software keyboard.
   [[UIKeyboardImpl sharedInstance] setAutomaticMinimizationEnabled:NO];
-  // Showing accessibility inspector turns on accessibility for the AUT.
-  [selfClass grey_setEnableAccessibilityInspector:YES];
-  // Turn off auto correction as it inteferes with typing on iOS8.2+.
+  // Enable accessibility.
+  [selfClass grey_setEnableAccessibility];
+  // Turn off auto correction as it interferes with typing on iOS8.2+.
   if (iOS8_2_OR_ABOVE()) {
     [selfClass grey_modifyKeyboardSettings];
   }
-
-  // Setup performed aboves leaves unwanted traces (such as the accessibility inspector)
-  // lingering around after all tests finish.
-  // Register to perform cleanup task right after the first test case ends. After much trial and
-  // error, this approach seems to work nicer than atexit and xctestobservation. atexit isn't
-  // reliable and gets called multiple times and using xctestobservation mysteriously eats up some
-  // of the logs.
-  [[NSNotificationCenter defaultCenter] addObserver:self
-                                           selector:@selector(grey_tearDown:)
-                                               name:kGREYXCTestCaseInstanceDidFinish
-                                             object:nil];
 }
 
 #pragma mark - Automation Setup
 
-// Enables (or disables) the accessibility inspector which is required for using any properties of
-// the accessibility tree.
-+ (void)grey_setEnableAccessibilityInspector:(BOOL)enabled {
-  NSString *accessibilitySettingsPrefBundlePath =
-      @"/System/Library/PreferenceBundles/AccessibilitySettings.bundle/AccessibilitySettings";
-  NSString *accessibilityControllerClassName = @"AccessibilitySettingsController";
-  id accessibilityControllerInstance =
-      [self grey_settingsClassInstanceFromBundleAtPath:accessibilitySettingsPrefBundlePath
-                                         withClassName:accessibilityControllerClassName];
-  [accessibilityControllerInstance setAXInspectorEnabled:@(enabled) specifier:nil];
+// Enables accessibility, which is required for using any properties of the accessibility tree.
++ (void)grey_setEnableAccessibility {
+  char const *const libAccessibilityPath =
+      [@"/usr/lib/libAccessibility.dylib" fileSystemRepresentation];
+  void *handle = dlopen(libAccessibilityPath, RTLD_LOCAL);
```

Comment:
I made a minor change here to dlopen/dlsym calls. 

With RTLD_LOCAL, if the dylib wasn't already loaded in global scope, the dynamic linker will not make its symbols available to anyone else. These symbols will then only be accessible through the handle provided by dlopen.

This way is better, because users won't be relying on EarlGrey dlopening this library, and it will be safe to remove this code in the future.